### ISSUE TITLE
[BUG]  Remove Pod Scheduling from harvester rke2 and rke1

### ISSUE THREAD

**Issue by WuJun2016** [harvester/harvester#2642]
**Describe the bug**
The presence of pod sheduling can be confusing for users

![image.png](https://images.zenhubusercontent.com/60345555ec1db310c78aa2b8/60cd2a5f-6675-41f1-89d8-244b5d391985)

![image.png](https://images.zenhubusercontent.com/60345555ec1db310c78aa2b8/48f520be-52cd-40b0-9f9e-dc30f501fa41)

**To Reproduce**

**Expected behavior**

**Support bundle**

**Environment**
 - Harvester ISO version: 
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

- https://github.com/harvester/harvester/issues/2316

**Comment by TachunLin** [comment]
To verify this change, we need to check the following:
1. `Pod Scheduling` did not appear on the RKE1 and RKE2 provisioning advanced options
2. The `Node Scheduling` function work as expected as https://github.com/harvester/harvester/issues/2316#issuecomment-1205046669

**Comment by TachunLin** [comment]
Verified **fixed** on Harvester `v1.0-44fb5f1a-head` (08/10) with Rancher `v2.6.7-rc7` Close this issue.

### Test Result
- [x] Pod Scheduling did not appear on the `RKE1` template provisioning advanced options
  ![image](https://user-images.githubusercontent.com/29251855/183801086-cc6de3d9-e420-42b5-af47-3ce7a60cd53c.png)
  
- [x] Pod Scheduling did not appear on the `RKE2` provisioning advanced options
  ![image](https://user-images.githubusercontent.com/29251855/183801181-67d24d86-fbb6-4bc8-9643-c7684a2406db.png)

- [x] **Can** deploy guest RKE1 cluster vm to specific Harvester node matching the `Node scheduling`
   ![image](https://user-images.githubusercontent.com/29251855/183807763-073b5622-5e77-4b23-989c-b610ff4d0586.png)
  ![image](https://user-images.githubusercontent.com/29251855/183807827-005bd590-5b24-4bf8-a26d-b81a4bbdc3dc.png)
  
- [x] **Can** deploy guest RKE2 cluster vm to specific Harvester node matching the `Node scheduling`
  ![image](https://user-images.githubusercontent.com/29251855/183804693-da78125c-0be8-4775-8665-37e0302877b9.png)
  ![image](https://user-images.githubusercontent.com/29251855/183804967-9e043d1a-cdc1-4233-a32e-63ea04b98125.png)
  
- [x] **Can** deploy guest RKE1 cluster vm to specific Harvester node matching the `Node scheduling` with `external kubconfig`

- [x] **Can** deploy guest RKE2 cluster vm to specific Harvester node matching the `Node scheduling` with `external kubconfig`

### Test resource 
We use the following ui-index url and node driver for testing:

**Rancher UI settings:** 
- ui-index: https://releases.rancher.com/ui/latest2/index.html (for RKE1)
- ui-dashboard-index: https://releases.rancher.com/dashboard/latest/index.html (for RKE2)
- ui-offline-preferred: Remote

**Node driver:**
- https://releases.rancher.com/harvester-node-driver/v0.5.0/docker-machine-driver-harvester-amd64.tar.gz

### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: `v1.0-44fb5f1a-head` (08/10)
* Rancher version: v2.6.7-rc7

### Environment Setup 
1. Prepare Harvester master node 
2. Prepare Rancher v2.6.7-rc7
3. Import Harvester to Rancher 
4. Set ui-offline-preferred: Remote
5. Go to Harvester Support page 
6. Download Kubeconfig 
7. Copy the content of Kubeconfig

### RKE2 Verification Steps
1. Open Harvester Host page then edit host config 
2. Add the following key value in the labels page:
  - topology.kubernetes.io/zone: zone_bp
  - topology.kubernetes.io/region: region_bp 
 ![image](https://user-images.githubusercontent.com/29251855/183802450-a790b9a2-3e2c-4559-8f84-b5a768b9c83d.png)
3. Open the RKE2 provisioning page 
4. Expand the show advanced 
5. Click add Node selector in `Node scheduling`
6. Use default `Required` priority
7. Click Add Rule
8. Provide the following key/value pairs
  - `topology.kubernetes.io/zone: zone_bp`
  - `topology.kubernetes.io/region: region_bp`
9. Provide the following user data 
  ```
  password: 123456
  chpasswd: { expire: False }
  ssh_pwauth: True
  ```
10. Create the RKE2 guest cluster
11. Go to Harvester Virtual Machine page
12. Edit yaml of the RKE2 guest cluster
  ![image](https://user-images.githubusercontent.com/29251855/183804372-776736cd-d58a-4f16-828f-45903d99af60.png)
13. Check the node affinity label have written into the yaml 
  ![image](https://user-images.githubusercontent.com/29251855/183804693-da78125c-0be8-4775-8665-37e0302877b9.png)
14. Check the guest cluster VM have no error message
  ![image](https://user-images.githubusercontent.com/29251855/183804967-9e043d1a-cdc1-4233-a32e-63ea04b98125.png)
15. Check can provision RKE2 cluster correctly
  ![image](https://user-images.githubusercontent.com/29251855/183805796-956e5ee0-129f-41b8-99e1-03764b6f436e.png)

### RKE1 Verification Steps
1. Follow the steps 1 ~ 7 of the RKE2 verification section
2. Go to Rancher Cluster Management page, add the RKE1 node template
3. Click add Node selector in `Node scheduling`
4. Use default `Required` priority
5. Click Add Rule
6. Provide the following key/value pairs
  - `topology.kubernetes.io/zone: zone_bp`
  - `topology.kubernetes.io/region: region_bp`
  ![image](https://user-images.githubusercontent.com/29251855/183807452-643faffb-f9f9-46b8-8414-5a8224aa3e66.png)
7. Create the RKE1 guest cluster
8. Go to Harvester Virtual Machine page
9. Edit yaml of the RKE1 guest cluster 
10. Check the node affinity label have written into the yaml 
  ![image](https://user-images.githubusercontent.com/29251855/183807763-073b5622-5e77-4b23-989c-b610ff4d0586.png)
11. Check the guest cluster VM have no error message
  ![image](https://user-images.githubusercontent.com/29251855/183807827-005bd590-5b24-4bf8-a26d-b81a4bbdc3dc.png)
12. Check can provision RKE1 cluster without error
  ![image](https://user-images.githubusercontent.com/29251855/183808305-d64c828f-7a08-4fc5-ac27-cbb879d57f2f.png)
